src/main/java/com/cairnfg/waypoint/authorization/endpoints/account/GetAllAccountsEndpoint.java

```diff
@@ -115,33 +116,62 @@ private ResponseEntity<AccountListDto> buildUnfilteredAccountList(String modifie
-                                .map(Role::getName)
-                                .toList())
-                            .build())
-                        .associatedAccounts(AssociatedAccountListDto.builder()
-                            .accounts(Stream.concat(Stream.of(account.getCoClient()), account.getDependents().stream())
-                                .map(Account.class::cast)
-                                .filter(Objects::nonNull)
-                                .map(associatedAccount -> AssociatedAccountDto.builder()
-                                    .id(associatedAccount.getId())
-                                    .username(associatedAccount.getUsername())
-                                    .firstName(associatedAccount.getFirstName())
-                                    .lastName(associatedAccount.getLastName())
-                                    .role(
-                                    	HouseholdHelperService.getHouseholdRole(account.getHousehold(), associatedAccount))
-                                    .build())
-                                .toList())
-                            .build())
-                        .build())
+                    .map(account -> {
+                      Household household = account.getHousehold();
+                      HouseholdRoleEnum householdRole = HouseholdRoleEnum.NONE;
+
+                      // Check for null household before calling the helper service
+                      if (household != null) {
+                        householdRole = HouseholdHelperService.getHouseholdRole(household, account);
+                      } else {
+                        log.warn("Account [{}] has no household associated.", account.getId());
+                      }
+
+                      return AccountDto.builder()
+                          .id(account.getId())
+                          .firstName(account.getFirstName())
+                          .lastName(account.getLastName())
+                          .email(account.getEmail())
+                          .householdId(household == null ? null : household.getId())
+                          .accountRoles(AccountRolesListDto.builder()
+                              .roles(account.getRoles().stream()
+                                  .map(Role::getName)
+                                  .toList())
+                              .build())
+                          .associatedAccounts(AssociatedAccountListDto.builder()
+                              .accounts(Stream.concat(
+                                      Stream.of(account.getCoClient()),
+                                      account.getDependents().stream()
+                                  )
+                                  .filter(Objects::nonNull)
+                                  .map(associatedAccount -> {
+                                    Household associatedHousehold = associatedAccount.getHousehold();
```

Comment:
I like the creative use of the streams here. This can get a bit confusing to follow. May want to refactor into a method call via Lambda (arrow function) later on